src/rust/engine/remote_provider/remote_provider_reapi/src/byte_store.rs

```diff
@@ -393,24 +395,48 @@ impl ByteStoreProvider for Provider {
         &self,
         digests: &mut (dyn Iterator<Item = Digest> + Send),
     ) -> Result<HashSet<Digest>, String> {
-        let request = remexec::FindMissingBlobsRequest {
-            instance_name: self.instance_name.as_ref().cloned().unwrap_or_default(),
-            blob_digests: digests.into_iter().map(|d| d.into()).collect::<Vec<_>>(),
-        };
+        let blob_digests = digests.into_iter().map(|d| d.into()).collect::<Vec<_>>();
+
+        let max_digests_per_request: usize = (4 * 1024 * 1024
+            - self
+                .instance_name
+                .as_ref()
+                .cloned()
+                .unwrap_or_default()
+                .len())
+            / RPC_DIGEST_SIZE;
 
-        let client = self.cas_client.as_ref().clone();
+        let requests = blob_digests.chunks(max_digests_per_request).map(|digests| {
+            let msg = remexec::FindMissingBlobsRequest {
+                instance_name: self.instance_name.as_ref().cloned().unwrap_or_default(),
+                blob_digests: digests.to_vec(),
+            };
+
+            msg
+        });
 
         workunit_store::increment_counter_if_in_workunit(Metric::RemoteStoreExistsAttempts, 1);
```

Comment:
I'm... not sure whether these should be counting `list_missing_digests` calls (i.e. remain here) or network requests (i.e. move within the `map(|request| { ... })` closure). 🤔